diff -Naur PandoraMonitoring-v03-05-00/src/TTreeWrapper.cc PandoraMonitoring-v03-05-00/src/TTreeWrapper.cc
--- PandoraMonitoring-v03-05-00/src/TTreeWrapper.cc	2020-06-26 03:59:42.000000000 -0500
+++ PandoraMonitoring-v03-05-00/src/TTreeWrapper.cc	2023-03-09 10:08:03.623846954 -0600
@@ -266,13 +266,6 @@
 //------------------------------------------------------------------------------------------------------------------------------------------
 
 template <typename T>
-TTreeWrapper::Branch<T>::~Branch()
-{
-}
-
-//------------------------------------------------------------------------------------------------------------------------------------------
-
-template <typename T>
 void TTreeWrapper::Branch<T>::Set(T variable)
 {
     m_variable = variable;
