index db92496..0135513 100644
--- a/mcs/basic_driver/CMakeLists.txt
+++ b/mcs/basic_driver/CMakeLists.txt
@@ -1,3 +1,5 @@
+if (${WANT_KMOD})
+if (EXISTS /lib/modules/${KVERS}/build)
 if (DEFINED ENV{KDIR})
   set(KDIR "$ENV{KDIR}")
   execute_process(COMMAND cat ${KDIR}/include/config/kernel.release OUTPUT_VARIABLE KVERS OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -34,5 +36,9 @@ add_custom_command(
 add_custom_target(pci_devel_module ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/../../drivers/${KVERS}/pci_devel_main.ko)
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../drivers DESTINATION .)
 
+else()
+        message(FATAL_ERROR "KMOD build requested, but kernel-devel package is not installed!")
+endif()
+endif()
 install_headers()
 install_source()
diff --git a/mu2e_driver/CMakeLists.txt b/mu2e_driver/CMakeLists.txt
index 3c54004..0ccbcd3 100644
--- a/mu2e_driver/CMakeLists.txt
+++ b/mu2e_driver/CMakeLists.txt
@@ -5,6 +5,8 @@ else()
   execute_process(COMMAND uname -r OUTPUT_VARIABLE KVERS OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 
+if (${WANT_KMOD})
+if (EXISTS /lib/modules/${KVERS}/build)
 #get_cmake_property(_variableNames VARIABLES)
 #foreach(_variableName ${_variableNames})
 #      message(STATUS "${_variableName}=${${_variableName}}")
@@ -58,6 +60,10 @@ add_custom_command(
 
 add_custom_target(mu2e_driver ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/../drivers/${KVERS}/mu2e.ko)
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../drivers DESTINATION .)
+else()
+   message(FATAL_ERROR "KMOD build requested, but kernel-devel package is not installed!")
+endif()
+endif()
 
 install_headers()
 install_source()
