--- spack-src/CMakeLists.txt	2024-04-09 17:03:45.358649348 -0500
+++ spack-src/CMakeLists.txt.orig	2024-03-21 11:19:54.000000000 -0500
@@ -2,19 +2,27 @@
 
 #  This simply copies files into an install directory - there is no build
 
-CMAKE_MINIMUM_REQUIRED (VERSION 3.19)
+CMAKE_MINIMUM_REQUIRED (VERSION 2.8)
 
-project(larbatch VERSION 5.15.51)
+project(larbatch NONE)
 
-find_package(cetbuildtools)
+# cetbuildtools contains our cmake modules
+set( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
+if( NOT CETBUILDTOOLS_VERSION )
+  message( FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules" )
+endif()
+set( CMAKE_MODULE_PATH $ENV{CETBUILDTOOLS_DIR}/Modules ${CMAKE_MODULE_PATH} )
 
 include(CetCMakeEnv)
 
 # larbatch has no architecture dependence
 cet_cmake_env( noarch )
 
+# Make sure cetbuildtools is set up (version is minimum)
+find_ups_product(cetbuildtools v4_09_00)
+
 # source code
-#add_subdirectory(ups)
+add_subdirectory(ups)
 add_subdirectory(scripts)
 add_subdirectory(python)
 
